src/common/containers/GameArea.js

```diff
@@ -291,19 +277,33 @@ export class GameArea extends Component {
     return this.props.player === 'blue' ? this.props.bluePieces : this.props.orangePieces;
   }
 
-  movePiece(hexId, asPartOfAttack = false) {
+  updateDimensions = (props = this.props) => {
+    const maxBoardHeight = window.innerHeight - 64 - 150;
+    const maxBoardWidth = window.innerWidth - (props.sidebarOpen ? 512 : 256);
+
+    this.setState({
+      areaHeight: window.innerHeight - 64,
+      boardSize: Math.min(maxBoardWidth, maxBoardHeight)
+    });
+  }
+
+  toggleChat = () => {
+    this.setState({ chatOpen: !this.state.chatOpen });
+  }
+
+  movePiece = (hexId, asPartOfAttack = false) => {
     this.props.onMoveRobot(this.props.selectedTile, hexId, asPartOfAttack);
   }
 
-  attackPiece(hexId, intermediateMoveHexId) {
+  attackPiece = (hexId, intermediateMoveHexId) => {
     if (intermediateMoveHexId) {
       this.props.onMoveRobotAndAttack(this.props.selectedTile, intermediateMoveHexId, hexId);
     } else {
       this.props.onAttackRobot(this.props.selectedTile, hexId);
     }
   }
 
-  placePiece(hexId) {
+  placePiece = (hexId) => {
```

Comment:
Is this how we're writing functions now?

Reply:
Yes, because only the arrow syntax preserves `this` correctly. That's what caused the bug.